spark/src/main/java/org/apache/comet/CometBatchIterator.java

```diff
@@ -45,12 +49,27 @@ public class CometBatchIterator {
    * indicating the end of the iterator.
    */
   public long[] next() {
+    // The native executor should have moved the previous batch, it is safe for us to deallocate
+    // the ArrowSchema and ArrowArray base structures.
```

Comment:
Not always. A native operator could possibly save batches internally.

Reply:
Yes, even though the native operator save batches internally, the batch would be moved to the native operator to be saved.